mat-process-api.Tests/V1/Controllers/ProcessDataImagesControllerTests.cs

```diff
@@ -175,5 +182,329 @@ public void given_an_exception_thrown_check_that_the_controller_throws_correct_e
+            _mockGetValidator.Setup(v => v.Validate(It.IsAny<GetProcessImageRequest>())).Returns(fakeValidationResult);
+
+            var expectedControllerResponse = new BadRequestObjectResult(validationErrorList); // build up expected controller response to check if the contents of the errors match - that's probably the easiest way to check that.
+
+            //act
+            var controllerResponse = _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+            var result = controllerResponse as ObjectResult;
+            var resultContents = (IList<ValidationFailure>)result.Value;
+
+            //assert
+            Assert.NotNull(controllerResponse);
+            Assert.NotNull(result);
+            Assert.IsInstanceOf<BadRequestObjectResult>(result);
+
+            Assert.IsInstanceOf<IList<ValidationFailure>>(resultContents);
+            Assert.NotNull(resultContents);
+            Assert.AreEqual(errorCount, resultContents.Count); // there should be 1 validation failure coming from the invalid guid
+            Assert.AreEqual(JsonConvert.SerializeObject(expectedControllerResponse), JsonConvert.SerializeObject(controllerResponse)); //expectedControllerResponse
+        }
+
+        [Test]
+        public void given_an_unexpected_error_would_be_thrown_when_GetProcessImage_controller_method_is_called_then_it_returns_500_status_code()
+        {
+            //arrange
+            int randomErrorNumber = _faker.Random.Int(); //triangulating unexpected exceptions
+            string messageRandomizer = _faker.Random.Hash().ToString(); //triangulating error messages
+
+            _mockUsecase.Setup(x => x.ExecuteGet(It.IsAny<GetProcessImageRequest>()))
+                .Returns(() => {
+                    ErrorThrowerHelper.GenerateError(randomErrorNumber, messageRandomizer); // throw that same random test error
+                    return new GetProcessImageResponse(null, DateTime.MinValue, null); //dummy return that will never happen due to error being thrown prior to it. It's only needed to keep compiler happy.
+                });
+
+            var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+            _mockGetValidator.Setup(x => x.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //validation successful
+
+            //act
+            var controllerResponse = _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
```

Comment:
**This is repeated comment, but same issue in different place:**

> Related to comment '<1>' . For this test the contents of the request don't matter, I think this controller method call should be supplied with nulls to indicate that to avoid confusion about the test. Also it makes test harder to read and follow.
> 
> Like that:
> `var controllerResponse = _processImageController.GetProcessImage(null, null, null ,null);`

Reply:
Obsolete due to model binding changes.